cfpb/teachers-digital-platform
# CR-Tool Business logic organization

Organized all business logic into a folder named "business.logic".  Updated the code to meet industry and organization conversions (where possible)

## Additions

- The below hieararchy
  - business.logic (folder)
    - repository.js (new location for all localStorage specific logic)
    - criterionService.js (new location for all Criterion specific logic)
    - dateTimeFormatter.js (new code that formats DateTime specific to this project)
    - constants.js (moved here)

## Changes

- Moved existing logic out of "CustomerReviewToolComponent.js" 

## Testing

- run `./setup.sh` in repo root
- load http://localhost:8000/tdp/crt-start/
- Enter Curriculum Title and choose a Grade Range (helps if you name the title similar to the choosen grade)
- click "Start Review"
- Verify the following logic still works for one or more of the "Dimensions (also called Distinctives)"
  - [ ] Answering all the required fields for a given Criterion (radio buttons for most dimensions) completes the Criterion (add's a green check to the left of the Criterion title
  - [ ] Criterion 2 and up are expandable links
  - [ ] Answering all the required fields for the entire Dimension enables the summary button
  - [ ] On the summary page (possibly not complete) There is a "View or edit response" link that takes you back to the criterion survey and puts the Dimension back to in progress
- Note: Efficacy is still a work in progress so it might not be clear that it is behaving correctly.

## Review

- @contolini @ascott1 @Scotchester @willbarton


[Preview this PR without the whitespace changes](?w=0)
